File: src/main/java/com/bso/dracko/mediator/springboot/EnableDrackoMediator.java
package com.bso.dracko.mediator.springboot;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.springframework.context.annotation.ComponentScan;

@ComponentScan(basePackages = "com.bso.dracko.mediator.springboot")
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface EnableDrackoMediator {

}

File: src/main/kotlin/com/bso/dracko/mediator/springboot/EnableDrackoMediator.kt
package com.bso.dracko.mediator.springboot

import org.springframework.context.annotation.ComponentScan

@ComponentScan(basePackages = ["com.bso.dracko.mediator.springboot"])
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class EnableDrackoMediator
